### PR TITLE
python3 migration

### PR DESCRIPTION
**Pull request by sansen**
Updated the code to be python 3 compatible.
I resolved some issues due to deprecated code or modules.

Related to #13 and #26

### CODE REVIEW

File: requirements_py3.txt [hunk 1]
```diff
@@ -1,6 +1,10 @@
...
-beautifulsoup4==4.6.0
-m3u8==0.3.2
-slimit==0.8.1
-ply==3.4
+beautifulsoup4==4.8.2
+certifi==2019.11.28
+chardet==3.0.4
+defer==1.0.4
+idna==2.9
+pkg-resources==0.0.0
```
Comment (facundobatista) [inline requirements_py3.txt]: This version is wrong

File: requirements_py3.txt [hunk 2]
```diff
@@ -1,6 +1,10 @@
-nose>=1.3.1
-yaswfp>=0.4
-beautifulsoup4==4.6.0
-m3u8==0.3.2
-slimit==0.8.1
-ply==3.4
```
Comment (facundobatista) [inline requirements_py3.txt:6]: Don't remove these dependencies! They are needed to run the tests. Try `./test` (which needs also to be updated, see that runs flake8 using python2).

File: requirements_py3.txt [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
+beautifulsoup4==4.8.2
+certifi==2019.11.28
+chardet==3.0.4
+defer==1.0.4
+idna==2.9
+pkg-resources==0.0.0
+pyxdg==0.26
+requests==2.23.0
+soupsieve==2.0
+urllib3==1.25.8
```
Comment (facundobatista) [inline requirements_py3.txt:17]: Beyond other comments in this file, this whole file needs to go away. We should have only one requirements file, for Python 3 (ideally, we should have 2, one for "production", the other for the extra dependencies to run the tests, but we can make that separation later).

File: encuentro/data.py [hunk 4]
```diff
@@ -294,15 +292,15 @@ def __setitem__(self, pos, value):
 
     def values(self):
         """Return the iter values of the data."""
-        return self.data.itervalues()
+        return iter(self.data.values())
```
Comment (facundobatista) [inline encuentro/data.py]: IIRC, no need to do the `iter()` here and in the other cases, it was just an optimization which is the default in Py3.

File: encuentro/data.py [hunk 5]
```diff
@@ -294,15 +292,15 @@ def __setitem__(self, pos, value):
...
 
     def __len__(self):
         """The length."""
         return len(self.data)
 
     def items(self):
         """Return the iter items of the data."""
-        return self.data.iteritems()
+        return iter(self.data.items())
```
Comment (facundobatista) [inline encuentro/data.py]: Same above.

File: encuentro/ui/main.py [hunk 6]
```diff
@@ -342,6 +340,8 @@ def show_message(self, err_type, text):
                     text = text.decode("latin1")
                 except UnicodeDecodeError:
                     text = repr(text)
+            except AttributeError:
+                pass
```
Comment (facundobatista) [inline encuentro/ui/main.py:345]: Never hide an error like this. I prefer it to explode and we will debug the problem.

File: encuentro/update.py [hunk 7]
```diff
@@ -214,7 +214,7 @@ def _merge(self, backends):
             final_data.append(d)
 
         logger.debug("Merging: appending other data: %s", backends.keys())
-        for data in backends.itervalues():
+        for data in iter(backends.values()):
```
Comment (facundobatista) [inline encuentro/update.py]: Same than above.

**Review by facundobatista — CHANGES_REQUESTED** [review]
Hey, great work!! Most of the changes here makes total sense.

I included some inline comments, the main blocking issue to do a final review is the problem with the requirements file.

Once that's solved, I'd be able to test run this, and check all the unicode/bytes conversion, which I need to see what we're having before confirming those changes make sense.

Thank you!!!!

File: requirements_py3.txt (hunk 2 above)
Comment (sansen) [reply requirements_py3.txt:6]: I've updated the dependecies and the test file for running ./test

File: requirements_py3.txt (hunk 1 above)
Comment (sansen) [reply requirements_py3.txt]: Removed it.

**Comment by facundobatista** [comment]
I'm merging this, and will apply real soon a set of other changes above.

**Review by facundobatista — APPROVED** [review]

**Pull request merged**